kernel/Kconfig

```diff
@@ -600,6 +600,13 @@ config SYSTEM_WORKQUEUE_NO_YIELD
 	  cooperative and a sequence of work items is expected to complete
 	  without yielding.
 
+config SYSTEM_WORKQUEUE_WORK_TIMEOUT_MS
+	int "Select system work queue work timeout in milliseconds"
+	default 10000 if DEBUG
+	default 0
```

Comment:
Zero could still be a valid setting? I'd argue you want a boolean option to conditionally compile in the feature, so zero overhead if the user does not need a timeout.

Reply:
0 needs to be valid since the feature is available to any workq, so if the feature is enabled, a value can be passed to `struct k_work_queue_config`

I can add a config to enable the work timeout, something like `config WORKQ_WORK_TIMEOUT` or something :)